vendor/bat-native-ads/src/bat/ads/internal/p2a/p2a.h

```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BAT_ADS_INTERNAL_P2A_P2A_H_
+#define BAT_ADS_INTERNAL_P2A_P2A_H_
+
+#include <stdint.h>
+
+#include <string>
+#include <vector>
+
+#include "base/values.h"
+
+namespace ads {
+
+class AdsImpl;
+
+class P2A {
+ public:
+  P2A(
+      AdsImpl* ads);
+
+  ~P2A();
+
+  void RecordEventForAdOpportunity(
+      const std::vector<std::string>& segments);
```

Comment:
P2A should not know about segments, can we no use:
```
  void RecordEvent(
      const std::string& name,
      const std::vector<std::string>& value);
```
and sanitize the segments outside of this class?

Reply:
Please remove as no longer required

Reply:
as discussed moved do `p2a/p2a_util.cc`